python/cookiecutter-django.md

```diff
@@ -0,0 +1,538 @@
+use_async [n]:
+use_drf [n]: y
+custom_bootstrap_compilation [n]: n
+use_compressor [n]:
+use_celery [n]:
+use_mailhog [n]: y
+use_sentry [n]:
+use_whitenoise [n]:
+use_heroku [n]: y
+Select ci_tool:
+1 - None
+2 - Travis
+3 - Gitlab
+Choose from 1, 2, 3 [1]:
+keep_local_envs_in_vcs [y]: n
+debug [n]:
+```
+
+</details>
+
+## Install pipenv
+The recommendation from Rootstrap is that you use `pipenv` to manage the development environment dependencies.
+
+To install the package, just run `$ pip install pipenv`.
+
+## Install dependencies
+To install the dev dependencies, run `$ pipenv install -r requirements/local.txt --dev`.
+
+NOTE:
+- Remember, when you install a new dependency, check if it will be needed at production or not. In the case that it won't be needed, add the flag `--dev` at the end of the installation command.
+- At pipenv to run a command, you must write before `pipenv run`, for example, the command that runs the server should be `$ pipenv run python manage.py runserver`.
+
+## Create psql database
+`$ createdb cookiecutter_starter-dev`
+
+## Load .env file for local configurations
+1. At `config/settings/base.py` set `DJANGO_READ_DOT_ENV_FILE` to load the configurations from `.env` file.
+```python
+#  config/settings/base.py
+
+#  Before
+READ_DOT_ENV_FILE = env.bool("DJANGO_READ_DOT_ENV_FILE", default=False)
+
+#  After
+READ_DOT_ENV_FILE = env.bool("DJANGO_READ_DOT_ENV_FILE", default=True)
+```
+
+2. Create `.env` file at root folder and `.env.example` to track the local configuration needed.
+```yml
+#  .env.example
+
+DJANGO_DEBUG=on
+DATABASE_URL=postgres://<user>:<password>@localhost:5432/cookiecutter_starter-dev
+```
+
+## Migrate
+`$ pipenv run python manage.py migrate`
+
+## Run test
+`$ pipenv run pytest`
+
+## Run test and get coverage percentage
+`$ pipenv run coverage run -m pytest && pipenv run coverage report -m`
+
+NOTE:
+- The command must run without exceptions because it mandatory for future configurations.
+- Don't matter if you write tests following the [unittest](https://docs.python.org/3/library/unittest.html) or [pytest](https://docs.pytest.org/) approach since `$ pipenv run pytest` command will be able to run both of them.
+
+## Install Mailhog
+`$ brew install mailhog`
+
+NOTE: If you prefer other options to install *Mailhog*, you can take a look at the [Mailhog installation guideline](https://github.com/mailhog/MailHog#installation).
+
+Start the server running `$ mailhog`
+
+Finally, if you go to `http://localhost:8025/`, your mail server is running.
+
+## Code Style
+### [flake8](https://flake8.pycqa.org/) and [flake8-isort](https://github.com/gforcada/flake8-isort)
+*NOTE: By default this package is already installed.*
+
+1. `$ pipenv install flake8 flake8-isort --dev`
+1. Add dependencies to requirement files. Below the *Code quality* separator:
+```
+flake8==<version>
+flake8-isort==<version>
+```
+3. Add custom configuration at `setup.cfg` file according to the Rootstrap's standards.
```

Comment:
Can we add here a link on `Rootstrap's standards`? or at least a file for example?